src/Microsoft.AspNetCore.Razor.Language/AllowedChildTagDescriptorComparer.cs

```diff
@@ -54,10 +54,9 @@ private AllowedChildTagDescriptorComparer(bool caseSensitive = false)
                 return false;
             }
 
-            return descriptorX != null &&
+            return
                 string.Equals(descriptorX.Name, descriptorY.Name, _stringComparison) &&
-                string.Equals(descriptorX.DisplayName, descriptorY.DisplayName, StringComparison.Ordinal) &&
-                Enumerable.SequenceEqual(descriptorX.Diagnostics, descriptorY.Diagnostics);
+                string.Equals(descriptorX.DisplayName, descriptorY.DisplayName, StringComparison.Ordinal);
```

Comment:
I think this was an old bug about us not using the `_stringComparison` parameter here. Might as well fix now since we're touching it anyhow.